pkg/operation/botanist/component/etcd/etcd.go

```diff
@@ -589,6 +595,11 @@ func (e *etcd) ServiceDNSNames() []string {
 func (e *etcd) SetSecrets(secrets Secrets)                 { e.secrets = secrets }
 func (e *etcd) SetBackupConfig(backupConfig *BackupConfig) { e.backupConfig = backupConfig }
 func (e *etcd) SetHVPAConfig(hvpaConfig *HVPAConfig)       { e.hvpaConfig = hvpaConfig }
+func (e *etcd) SetETCDConfig(config *configv1alpha1.ETCDConfig) {
+	if config != nil {
```

Comment:
Why is this `if` check needed? Remove it. As a user, if `config` is nil then I expect `e.etcdConfig` also to be set to `nil`.

Reply:
Removed